src/client/websocket/WebSocketShard.js

```diff
@@ -475,12 +475,19 @@ class WebSocketShard extends EventEmitter {
       return;
     }
     const hasGuildsIntent = new Intents(this.manager.client.options.intents).has(Intents.FLAGS.GUILDS);
-    // Step 2. Create a 15s timeout that will mark the shard as ready if there are still unavailable guilds
+    // Step 2. Create a timeout that will mark the shard as ready if there are still unavailable guilds
+    // * the timeout should be 15 seconds by default
+    // * this can be optionally be changed in the client options via fetchGuildTimeout
+    // * a timeout time of zero will skip this timeout
+
+    const guildTimeoutAmount = this.manager.client.options.fetchGuildTimeout || 1500;
+
     this.readyTimeout = setTimeout(
       () => {
         this.debug(
           `Shard ${hasGuildsIntent ? 'did' : 'will'} not receive any more guild packets` +
-            `${hasGuildsIntent ? ' in 15 seconds' : ''}.\n   Unavailable guild count: ${this.expectedGuilds.size}`,
+            `${hasGuildsIntent ? ` in ${guildTimeoutAmount} ms` : ''}.\n` +
+            `Unavailable guild count: ${this.expectedGuilds.size}`,
```

Comment:
I think this isn't allowed either. Just format with prettier and it will adjust it for you

Reply:
Thank you, i will try this in a little bit.

Reply:
I've used the prettier version! it looks so much better :) thank you